### PR TITLE
Cleanup: less confusion between config priority and balancing priority in connection pools

### PR DESCRIPTION
**Pull request by devcrafter**
Function/variable names were creating confusion between config priority and balancing priority

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit b85a68790aa37b06b8ce3cfe80b9e232315053a9 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60077/b85a68790aa37b06b8ce3cfe80b9e232315053a9/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/60077/b85a68790aa37b06b8ce3cfe80b9e232315053a9/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Client/ConnectionPoolWithFailover.cpp [hunk 1]
```diff
@@ -79,14 +79,6 @@ IConnectionPool::Entry ConnectionPoolWithFailover::get(const ConnectionTimeouts
...
 }
 
-Priority ConnectionPoolWithFailover::getPriority() const
-{
-    return (*std::max_element(nested_pools.begin(), nested_pools.end(), [](const auto & a, const auto & b)
-    {
-        return a->getPriority() < b->getPriority();
-    }))->getPriority();
-}
-
```
Comment (yakov-olkhovskiy) [inline src/Client/ConnectionPoolWithFailover.cpp:89]: we don't need this function?

Comment (devcrafter) [reply src/Client/ConnectionPoolWithFailover.cpp:89]: This `getPriority()` is about getting a cluster node priority from config. I can't picture how this method can be helpful for `ConnectionPoolWithFailover`. If anybody needs to get a cluster node with the highest priority in config then this functionality should be just a separate function and there is no need for pure virtual method in `IConnectionPool` for this

**Review by yakov-olkhovskiy — APPROVED** [review]

**Comment by devcrafter** [comment]
AST Fuzzer [error report](https://s3.amazonaws.com/clickhouse-test-reports/60077/b6277d5e87367c2eda974ca09f1c174563ab80d6/ast_fuzzer__asan_.html) should be fixed by https://github.com/ClickHouse/ClickHouse/pull/60227

**Pull request merged**